extensions/compose/src/extension.ts

```diff
@@ -50,7 +89,44 @@ async function postActivate(extensionContext: extensionApi.ExtensionContext): Pr
 }
 
 export async function deactivate(): Promise<void> {
+  stopMonitoringLoop = true;
   if (composeExtension) {
     await composeExtension.deactivate();
   }
 }
+
+// Monitor the provider and check to see if the binary has been installed system-wide or not,
+// if it has, then we can update the status to "installed" and the user can use the "podman compose" command
+async function monitorProvider(
+  provider: extensionApi.Provider,
+  extensionContext: extensionApi.ExtensionContext,
+): Promise<void> {
+  const os = new OS();
+  const detect = new Detect(os, extensionContext.storagePath);
+
+  // Call again
+  if (!stopMonitoringLoop) {
+    try {
+      // Detect that it's installed systemwide and update the status
+      const isDockerComposeInstalledSystemWide = await detect.checkSystemWideDockerCompose();
```

Comment:
as you have a monitor to check if docker is installed you could update the setting value with it.
if i have compose installed and i remove it, i would see the configuration value updates.

Reply:
Tried that and unfortunately it prevented even clicking on the toggle, as sometimes it'd update during the "install system wide" process when clicking the button (since it asks for the user's password + takes a few seconds for it to install).

So I chose not to also update the context values. Instead I have it double check the setting values when going through the "onboarding" process here in a different PR: https://github.com/containers/podman-desktop/pull/4179/commits/cc23037f40c93f13fd4479384d755bd5b428e06b#diff-7327606be194993bf69f195205c8cd4dc49b1cd805ad524caabe1163dcd51b39R71